mayastor/src/subsys/mbus/mbus_nats.rs

```diff
@@ -0,0 +1,100 @@
+//! NATS implementation of the `MessageBus` connecting mayastor to the control
+//! plane components.
+
+use super::MessageBus;
+use async_trait::async_trait;
+use nats::asynk::Connection;
+use once_cell::sync::OnceCell;
+use smol::io;
+
+pub(super) static NATS_MSG_BUS: OnceCell<NatsMessageBus> = OnceCell::new();
+pub(super) fn message_bus_init(server: String) {
+    NATS_MSG_BUS.get_or_init(|| {
+        // Waits for the message bus to become ready
+        tokio::runtime::Runtime::new()
+            .unwrap()
+            .block_on(async { NatsMessageBus::new(&server).await })
+    });
+}
+
+// Would we want to have both sync and async clients?
+pub struct NatsMessageBus {
+    connection: Connection,
+}
+impl NatsMessageBus {
+    pub async fn connect(server: &str) -> Connection {
+        info!("Connecting to the nats server {}...", server);
+        // We retry in a loop until successful. Once connected the nats
+        // library will handle reconnections for us.
+        let interval = std::time::Duration::from_millis(500);
+        let mut log_error = true;
+        loop {
+            match nats::asynk::connect(server).await {
+                Ok(connection) => {
+                    info!(
+                        "Successfully connected to the nats server {}",
+                        server
+                    );
+                    return connection;
+                }
+                Err(error) => {
+                    if log_error {
+                        warn!(
+                            "Error connection: {}. Quietly retrying...",
+                            error
+                        );
```

Comment:
If I understand correctly we will have two of these warning messages per second printed to log file if nats is not available. Perhaps it would have been better to print it only once (the first time).

Reply:
hmm, actually it only logs it once, see the log_error bool?

Reply:
ah yes. thanks 👍